### DISCUSSION TITLE
How to attach screenshots after not-Selenide assertions (JUnit in my case) to Allure Report automatically?

### DISCUSSION THREAD

**Discussion by VladislavShcherba** [selenide/selenide#2827] [Q&A]
I know that Selenide takes screenshot on test failure by default. But it does it only for Selenide's assertions like `shouldBe`.
To solve this problem it is possible to add `@ExtendWith(ScreenShooterExtension.class)` to tests superclass. Now Selenide takes screenshots after not-Selenide (JUnit in my case) assertions.

I also want to use Allure Report. In order to integrate it with Selenide I use [allure-selenide](https://mvnrepository.com/artifact/io.qameta.allure/allure-selenide) library and add

```
    @BeforeAll
    public static void setupAllureReports() {
        SelenideLogger.addListener("AllureSelenide", new AllureSelenide()
                .screenshots(true)
                .savePageSource(false)
        );
    }
   
 ```
 to tests superclass.
 
As a result the Selenide's assertions screenshots are taken and attached to Allure. The non-Selenide assertions screenshots are taken but are not attached to Allure.
 
 I wasn't able to find a built-in solution for that, so I ended up in creating my own JUnit extension.
 
 But I wonder if it's possible to tell Selenide to attach Junit assertions screenshots to Allure easily?

**Comment by asolntsev** [comment]
This is rather a question to allure-selenide library. 
I see from its source code that they don't use Selenide screenshots at all. Instead, they take their own screenshots.
